framework/components/ecl-galleries/ecl-gallery-dialog/ecl-gallery-dialog.config.js

```diff
@@ -0,0 +1,132 @@
+module.exports = {
+  title: 'Gallery - dialog (overlay)',
+  label: 'Gallery - dialog (overlay)',
+  preview: '@preview-center-transparent',
+  status: 'ready',
+  context: {
+    _demo: {
+      scripts: `
+        // This is only for demo purposes to facilitate end-users.
+
+        // Create a link.
+        var link = document.createElement('a');
+        var text = document.createTextNode('Open dialog');
+
+        // Include textual content.
+        link.appendChild(text);
+        link.title = "Click to test the modal";
+
+        // Add necessary demo attributes.
+        link.setAttribute('href', '#dialog');
+        link.setAttribute('class', 'ecl-link');
+        link.setAttribute('data-ecl-dialog', 'ecl-dialog');
+
+        // Show the link
+        document.body.append(link);
+
+        document.addEventListener('DOMContentLoaded', function () {
+            ECL.dialogs();
+            ECL.carousels();
+        });
+      `,
+    },
+    dialog_settings: {
+      extra_attributes: [
+        { name: 'aria-labelledby', value: 'dialog-title' },
+        { name: 'aria-describedby', value: 'dialog-description' },
+        { name: 'id', value: 'ecl-dialog' },
+        { name: 'aria-hidden', value: 'true' },
+      ],
```

Comment:
The link is created manually, as I think the trigger of a dialog/overlay should be independent from the js behavior. Meaning, the trigger could be a link, a button, or any other element with the attributes necessary for it to be a trigger. And because a trigger link is one example, I created it with JS.